File: native/mixdog-graph/tests/fixtures/parity/ruby/sample.rb
# Parity fixture: declaration and import shapes the graph reports for Ruby.
require 'json'
require_relative 'helper'

module Acme
  LIMIT = 10

  class Store
    attr_reader :name

    def initialize(name)
      @name = name
    end

    def read(key)
      JSON.generate(name: @name, key: key)
    end

    def name=(value)
      @name = value
    end

    def self.build(name)
      new(name)
    end

    class Inner
      def ping
        :pong
      end
    end
  end

  def self.helper
    Store.build('helper')
  end
end

def top_level(value)
  value.to_s
end

File: native/mixdog-graph/tests/fixtures/parity/lua/sample.lua
-- Parity fixture: declaration and import shapes the graph reports for Lua.
local json = require("dkjson")
local helper = require 'lib.helper'

local M = {}

function M.build(name)
  return setmetatable({ name = name }, { __index = M })
end

function M:read(key)
  return json.encode({ name = self.name, key = key })
end

local function normalize(value)
  return helper.trim(value)
end

function top_level(value)
  return normalize(value)
end

return M
